lib/scripts/mtw.js

```diff
@@ -292,6 +301,36 @@ export class ContentScript {
   }
 
   /**
+   * Translate one word in each sentence for a  paragraph.
+   * @param {Object} paragraph - Paragraph nodeType
+   * @param {Object} filteredTMap - filtered translation map
+   * @param {Object} iMap - HTML element for each translated word
+   */
+  translateOneWord(paragraph, filteredTMap, iMap) {
+    for (let i in paragraph.childNodes) {
+      if (paragraph.childNodes[i].nodeType === 3) {
+        if(!/^\s*$/.test(paragraph.childNodes[i].textContent)){
+          let sentences = paragraph.childNodes[i].textContent.split('.');
+          for (let j in sentences) {
+            let words = sentences[j].split(' ');
+            words = _.shuffle(words);
+            for (let k in words) {
+              if (filteredTMap[words[k]]) {
+                let x = sentences[j].replace(words[k], iMap[filteredTMap[words[k]]]);
+                sentences[j] = x;
+                break;
+              }
+            }
+          }
+          var newNode = document.createElement('span');
+          newNode.innerHTML = sentences.join('.');
```

Comment:
> Also, splitting words in a sentence may introduce errors due to rogue spaces. Hence I chose to sentence wise.

I agree. Could you add a comment in the code explaining this choice?

> This is equivalent (or rather almost equivalent) to randomly replacing one word in a sentence.

What we have now is "at most one word per sentence", and not "one word per sentence".  It is important to be precise about this in the UI, otherwise some users will think that MTW has bugs. Could you update the UI accordingly?